complaint_search/decorators.py

```diff
@@ -1,18 +1,31 @@
+import logging
 from rest_framework import status
 from rest_framework.response import Response
 from elasticsearch import TransportError
 
+log = logging.getLogger(__name__)
+
+
 def catch_es_error(function):
     def wrap(request, *args, **kwargs):
         try:
             return function(request, *args, **kwargs)
-        except TransportError as e:
-            status_code = e.status_code if isinstance(e.status_code, int) \
-                else status.HTTP_400_BAD_REQUEST
+        except TransportError as te:
```

Comment:
I'm not sure I see the utility in catching this exception separately and returning a `4xx` here. If ES is having trouble, that still seems like a backend problem, so `5xx` would appear appropriate.

And now that we're logging the exceptions, it doesn't seem like we'd need to say anything about ES in the error response, and could just go with your more generic `There was a problem retrieving your request`. My vote would be for removing this `except` block and just keeping the blanket `except` and `500` response you have got below it.

Also, I may be reading the specs incorrectly, but it seems possible `424`s are only intended for WebDAV?

Reply:
IIRC, When we used a 500 response, the browser/Javascript performed some additional actions that we did not like.  I can't remember offhand (it's been over a year), but we had to use 4XX errors to maintain the user experience.  I can research this further, but for now, I would like to keep the ES not responding as separate from "out of disk space" types of catastrophies

Reply:
It'd be nice from a maintenance point of view to have two separate errors - that would simplify debugging if we know that one is "ES is down" and another is "something is badly wrong".